regen/regcomp.pl

```diff
@@ -701,6 +701,15 @@ sub print_reg_intflags_name {
                 chomp($next_line);
                 $line .= $next_line;
             }
+            if ( $line =~ m!/\*! ) {
+                while ( $line !~ m!\*/!) {
+                    defined(my $next_line= <$in_fh>)
+                        or last;
+                    chomp($next_line);
+                    $next_line=~s/^\s*/ /;
```

Comment:
Good questions, I can update the code with a comment to explain, and harden the code further to deal with star style comments, although in practice we don't use them on these defines it might save someone some surprise in the future. Below is some context:
  
<details>

The code in question is trying to parse out defines to build up documentation and introspection datasets in regnodes.h so we can show the flags that are set on a pattern in re 'debug' output. If you do a `git grep NAUGHTY` you will see a good example. The loop in question is looking for things like this:

```
#define PREGf_XXX 0x100 /* the xxx flags */
```

And it wants to turn into a tuple like this `["XXX", 0x100, "the xxx flags"]` for further processing.

However if and when the comment gets wrapped to this:

```
#define PREGf_XXX 0x100 /* the xxx 
                           flags */
```

The old logic would fail to match entirely and the given flag. With the new logic we would handle it correctly, first by noticing the define at all, and second by unwrapping the comment. You are correct that if the comments were "star style comments" that the unwrap wouldn't work. 

But currently they are all one line, the only reason they get wrapped are they are overlong, and my script wraps them and when it does so it does not add stars, so this change armors against that wrapping to come in the PR. It wouldn't hurt to armor this logic against start style comments as well, so i will add that, but strictly speaking it isn't necessary for this as the comments in question already exist and dont use that style.

Just FYI, regen/regcomp.pl takes the node and state definition data in regcomp.sym and combines it with metadata from a few of the other regex header files, and then produces regnodes.h, which is what the regex engine actually uses for processing. PREGf flags are special flags that allow the regex engine to determine how to run the pattern and some subtleties like that. For instance PREGf_GPOS is set if the regex engine detects that a pattern might go hyperexponential in performance (IIRC). 

```
$ ./perl -Ilib -Mre=Debug,DUMP,FLAGS -e'qr/\Gfoo/'
Compiling REx "\Gfoo"
Final program:
   1: GPOS (2)
   2: EXACT <foo> (4)
   4: END (0)
anchored "foo" at 0..0 (checking anchored noscan) anchored(GPOS) GPOS:0 minlen 3 
r->extflags: USE_INTUIT_NOML USE_INTUIT_ML IS_ANCHORED 
r->intflags: NOSCAN GPOS_FLOAT RECURSE_SEEN 
Freeing REx: "\Gfoo"
```
the logic we are discussing is required output the `r->intflags` and `r->extflags` lines.
</details>